src/langchain_google_alloydb_pg/vectorstore.py

```diff
@@ -613,28 +763,48 @@ async def __query_collection(
             results = await self.engine._afetch(stmt)
         return results
 
+    def embedding_helper(
+        self,
+        query: Optional[str] = None,
+        image_uri: Optional[str] = None,
+    ) -> List[float]:
+        if query and image_uri:
+            raise ValueError(
+                f"`query` and `image_uri` cannot be specified at the same time."
+            )
+        elif query:
+            embedding = self.embedding_service.embed_query(text=query)
+        elif image_uri:
+            embedding = self.embedding_service.embed_image([image_uri])[0]
```

Comment:
Most integrations' `embed_image` API accept a list of input images and returns a list of image embeddings (although the API name is singular). Here we are only embedding 1 input image_uri instead of a list, so we want to return the item in the list by indexing.

Reply:
Ah, another interface issue. This doesn't work with VertexAIEmbeddings either. 
@kurtisvg is there a better way to try to support services that take either a list or a single string than a try/except statement?

Reply:
added try/except statement for now